pkg/cmd/kafka/acl/grant/grant.go

```diff
@@ -0,0 +1,228 @@
+package grant
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/redhat-developer/app-services-cli/internal/config"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/kafka/acl"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+	"github.com/spf13/cobra"
+
+	kafkainstanceclient "github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal/client"
+)
+
+type options struct {
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     logging.Logger
+	IO         *iostreams.IOStreams
+	localizer  localize.Localizer
+	Context    context.Context
+
+	kafkaID    string
+	topic      string
+	user       string
+	svcAccount string
+	group      string
+	producer   bool
+	consumer   bool
+	prefix     bool
+	admin      bool
+}
+
+// NewGrantPermissionsACLCommand creates a series of ACL rules
+func NewGrantPermissionsACLCommand(f *factory.Factory) *cobra.Command {
+
+	opts := &options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Context:    f.Context,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "grant-permissions",
+		Short:   f.Localizer.MustLocalize("kafka.acl.grantPermissions.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("kafka.acl.grantPermissions.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("kafka.acl.grantPermissions.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasKafka() {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.noKafkaSelected")
+			}
+
+			opts.kafkaID = cfg.Services.Kafka.ClusterID
+
+			if opts.user == "" && opts.svcAccount == "" {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.noPricipalsSelected")
+			}
+
+			if opts.user != "" && opts.svcAccount != "" {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.bothPricipalsSelected")
+			}
+
+			return runGrantPermissions(opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.user, "user", "", opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.user.description"))
+	cmd.Flags().StringVar(&opts.svcAccount, "service-account", "", opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.serviceAccount.description"))
```

Comment:
We need to be clear that the service account client ID is used for this, not the service account ID (confusing as hell having two).

Could we merge these into a single flag like the UI does?

Reply:
btw.. validation is not required immediately/right now and can be added later phase over released commands